xen/include/public/arch-arm.h

```diff
@@ -318,6 +318,10 @@ struct xen_arch_domainconfig {
      *
      */
     uint32_t clock_frequency;
+#define _XEN_ARM_FLAGS_FFA        0
+#define XEN_ARM_FLAGS_FFA         (1U << _XEN_ARM_FLAGS_FFA)
+    /* IN */
+    uint32_t flags;
```

Comment:
You are introducing a generic value "flags" in a public header.
I am wondering if it would not make more sense to add a:
#define XEN_DOMCTL_CONFIG_TEE_FFA 2
At the end having optee support and FFA support in the same guest does not make sense

Reply:
While I agree that it would be nice to avoid adding another field it becomes a bit confusing.
How about leaving this as is and discussing it on the mailing list when posted?

Reply:
If you look in the mailing list you will see that there is a serie to add SVE which is ending in the same kind of question.
What we can do is keep it as is for now and wait for the decision on the SVE serie and then do exactly as decided for SVE :-)

Reply:
as stated before, we should definitely use tee_type instead and introduce a new value for it